### PR TITLE
Adding protocolhandlers

### PR DESCRIPTION
**Pull request by diekus**
Closes #863

This change (choose at least one, delete ones that don't apply):

* Breaks existing normative behavior (please add label "breaking")
* [x] Adds new normative recommendations or optional items
* [x] Makes only editorial changes (only changes informative sections, or changes normative sections without changing behavior)
* Is a "chore" (metadata, formatting, fixing warnings, etc).

Implementation commitment (delete if not making normative changes):

* [ ] Safari ([link to issue](https://bugs.webkit.org/show_bug.cgi?id=211401))
* [x] Chrome ([link to issue](https://bugs.chromium.org/p/chromium/issues/detail?id=1019239))
* [ ] Firefox ([link to issue](https://bugzilla.mozilla.org/show_bug.cgi?id=1635272))
* [x] Edge ([public signal](https://github.com/MicrosoftEdge/MSEdgeExplainers/blob/main/URLProtocolHandler/explainer.md))

If change is normative, and it adds or changes a member:

* [ ] [updated JSON schema](https://github.com/SchemaStore/schemastore/blob/master/src/schemas/json/web-manifest.json)

Commit message:

Adds "protocol_handlers" member to the manifest spec, referencing the process that navigator.registerProtocolHandler uses.

contributors:
+@fabiorocha

Person merging, please make sure that commits are squashed with one of the following as a commit message prefix:

* chore:
* editorial:
* BREAKING CHANGE:
* And use none if it's a normative change

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/manifest/pull/969.html" title="Last updated on Apr 12, 2021, 6:17 PM UTC (d1c606c)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/manifest/969/8c7fd7b...d1c606c.html" title="Last updated on Apr 12, 2021, 6:17 PM UTC (d1c606c)">Diff</a>

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -2708,6 +2824,91 @@ <h3>
...
+        <p>
+          When a <a>ProtocolHandlerItem</a> <var>protocol_handler</var> having
+          <a>WebAppManifest</a> <var>manifest</var> is invoked, run the
+          following steps:
+        </p>
+        <ol>
+          <li>Let <var>url</var> be <var>protocol_handler.url</var>.
+          </li>
+          <li>Replace the first occurrence of the exact literal string "%s" in
+          <var>url</var> with an escaped version of the absolute URL.
```
Comment (marcoscaceres) [inline index.html:2901]: The "an escaped version" needs some definition (or link to) of how to do the escaping.

**Review by marcoscaceres — CHANGES_REQUESTED** [review]
Thanks for taking first stab at this @diekus. 

Few things: we've dropped IDL from the spec, so be good to update to match the updated style. Similarly, we don't have "issue a developer warning" anymore.

There also need to be some strong security assurances around those URL replacements. It might be worth looking at how Web Share Target handles that (or perhaps some other specs that do something similar).

File: index.html (hunk 1 above)
Comment (diekus) [reply index.html:2901]: Oh yes, I see it's a completely different structure in general. I'll update it to be using this new fresh-chic style. :) brb

**Comment by diekus** [comment]
I'm closing this PR since I've noticed how far behind it is and it's simpler to branch from the current state of main for now and apply the new changes there..

**Comment by jayr2trilly** [comment]
> Closes #863
> 
> This change (choose at least one, delete ones that don't apply):
> 
> * Breaks existing normative behavior (please add label "breaking")
> * [x]  Adds new normative recommendations or optional items
> * [x]  Makes only editorial changes (only changes informative sections, or changes normative sections without changing behavior)
> * Is a "chore" (metadata, formatting, fixing warnings, etc).
> 
> Implementation commitment (delete if not making normative changes):
> 
> * [ ]  Safari ([link to issue](https://bugs.webkit.org/show_bug.cgi?id=211401))
> * [x]  Chrome ([link to issue](https://bugs.chromium.org/p/chromium/issues/detail?id=1019239))
> * [ ]  Firefox ([link to issue](https://bugzilla.mozilla.org/show_bug.cgi?id=1635272))
> * [x]  Edge ([public signal](https://github.com/MicrosoftEdge/MSEdgeExplainers/blob/main/URLProtocolHandler/explainer.md))
> 
> If change is normative, and it adds or changes a member:
> 
> * [ ]  [updated JSON schema](https://github.com/SchemaStore/schemastore/blob/master/src/schemas/json/web-manifest.json)
> 
> Commit message:
> 
> Adds "protocol_handlers" member to the manifest spec, referencing the process that navigator.registerProtocolHandler uses.
> 
> contributors: +@fabiorocha
> 
> Person merging, please make sure that commits are squashed with one of the following as a commit message prefix:
> 
> * chore:
> * editorial:
> * BREAKING CHANGE:
> * And use none if it's a normative change
> 
> [Preview](https://pr-preview.s3.amazonaws.com/w3c/manifest/pull/969.html) | [Diff](https://pr-preview.s3.amazonaws.com/w3c/manifest/969/8c7fd7b...d1c606c.html)

**Pull request closed without merging**